homeassistant/components/onewire/sensor.py

```diff
@@ -121,6 +244,12 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sensor."""
+        if "count" in self._unit_of_measurement:
+            return int(self._state)
+        if "short" in self._unit_of_measurement:
+            return STATE_PROBLEM if int(self._state) else STATE_OK
+        if "present" in self._unit_of_measurement:
+            return STATE_ON if int(self._state) else STATE_OFF
```

Comment:
The last two looks like they should be binary sensors instead.

Reply:
I was looking at doing this initially, but it raises some questions.  Currently onewire is not a toplevel platform, it's a child of "sensor".  So the question is do I break out and duplicate the code into a binary_sensor for those two subdevices, or do we have to convert onewire into a toplevel platform so it can have both sensors and binary_sensors?

Or am I missing something and there is another way to do that?

Reply:
Common logic should move to a component module `__init__.py`, yes.

There are no top level platforms. Platforms are the modules that create entities for the different entity integrations. Platforms are always part of a device/service specific integration. Some integrations only provide a single platform and have all the logic inside the platform, ie the current state here. If more platforms are required, the common logic should be located in the `__init__.py` module of the integration.

Reply:
It's ok to wait with this refactor and not implement these two sensor types.